src/graph/GoExecutor.cpp

```diff
@@ -1356,39 +1357,56 @@ OptVariantType GoExecutor::VertexHolder::get(VertexID id, TagID tid,
     return value(std::move(res));
 }
 
-SupportedType GoExecutor::VertexHolder::getType(VertexID id, TagID tid, const std::string &prop) {
-    auto iter = data_.find(id);
-    if (iter == data_.end()) {
-        return getDefaultPropType(tid, prop);
-    }
-
-    auto iter2 = iter->second.find(tid);
-    if (iter2 == iter->second.end()) {
-        return getDefaultPropType(tid, prop);
+void GoExecutor::VertexHolder::add(
+    const std::vector<storage::cpp2::QueryResponse> &responses) {
+    size_t num = 0;
+    for (auto& resp : responses) {
+        auto *vertices = resp.get_vertices();
+        auto *vertexSchema = resp.get_vertex_schema();
+        if (vertices == nullptr || vertexSchema == nullptr) {
+            continue;
+        }
+        for (auto &vdata : *vertices) {
+            num += vdata.tag_data.size();
+        }
     }
 
-    return std::get<0>(iter2->second)->getFieldType(prop).type;
-}
+    data_.reserve(num);
 
-void GoExecutor::VertexHolder::add(const storage::cpp2::QueryResponse &resp) {
-    auto *vertices = resp.get_vertices();
-    if (vertices == nullptr) {
-        return;
-    }
+    for (auto& resp : responses) {
+        auto *vertices = resp.get_vertices();
+        auto *vertexSchema = resp.get_vertex_schema();
+        if (vertices == nullptr || vertexSchema == nullptr) {
+            continue;
+        }
 
-    auto *vertexSchema = resp.get_vertex_schema();
-    if (vertexSchema == nullptr) {
-        return;
-    }
-    for (auto &vdata : *vertices) {
-        std::unordered_map<TagID, VData> m;
-        for (auto &td : vdata.tag_data) {
-            DCHECK(td.__isset.data);
-            auto it = vertexSchema->find(td.tag_id);
-            DCHECK(it != vertexSchema->end());
-            m[td.tag_id] = {std::make_shared<ResultSchemaProvider>(it->second), td.data};
-        }
-        data_[vdata.vertex_id] = std::move(m);
+        std::transform(
```

Comment:
Em...  Mostly, each response has the same vertexSchema returned.  
Do we really need to do the transform for every response?